app/src/main/java/com/humberto/tasky/core/database/dao/TaskDao.kt

```diff
@@ -13,6 +14,9 @@ interface TaskDao {
     @Upsert
     suspend fun upsertTasks(taskEntities: List<TaskEntity>)
 
+    @Query("SELECT * FROM taskentity WHERE time BETWEEN :startOfDay AND :endOfDay")
+    fun getTasksForDay(startOfDay: Long, endOfDay: Long): Flow<List<TaskEntity>>
+
     @Query("SELECT * FROM taskentity WHERE id=:id")
     fun getTask(id: String): TaskEntity
```

Comment:
You might be getting that `IllegalStateException` because some of your DAO functions aren’t marked as suspending. Those functions are not main-safe. When you make them suspending, Room will move the execution off the main thread with coroutines.